frame/evm/src/lib.rs

```diff
@@ -821,32 +821,14 @@ impl<T: Config> Pallet<T> {
 	/// Remove an account.
 	pub fn remove_account(address: &H160) {
 		if <AccountCodes<T>>::contains_key(address) {
-			// Remember to call `dec_sufficients` when clearing Suicided.
-			<Suicided<T>>::insert(address, ());
```

Comment:
Not sure about this change. How will this affect the heuristics of previous "corrupted" suicided contracts?

Reply:
Ignore my previous comment, but we can probably get rid of all the logic related to `clear_suicided_contract`.

Reply:
Yes, with the new `selfdestruct` changes, we no longer need the suicide contract logic. However, we cannot remove all related logic before ensuring that current suicided contracts and their storage are properly removed using the storage-cleaner precompile, and that the corrupted suicided contract is also cleaned through the lazy migration extrinsic.